relint/parse.py

```diff
@@ -140,13 +144,17 @@ def print_culprits(matches, args):
 
     if args.summarize:
         for test, filenames in match_groups.items():
-            hint = Panel(
-                Markdown(test.hint, justify="left"),
-                title="Hint:",
-                title_align="left",
-                padding=(0, 2),
-            )
-            group = Group(Group(*filenames), hint)
+            if test.hint is not None:
+                hint = Panel(
+                    Markdown(test.hint, justify="left"),
+                    title="Hint:",
+                    title_align="left",
+                    padding=(0, 2),
+                )
+            if test.hint is None:
+                group = Group(*filenames)
+            else:
+                group = Group(Group(*filenames), hint)
```

Comment:
You have plenty of if statements here. I believe you can simply combine it, with the conditional statement above.